File: problems/2042-check-if-numbers-are-ascending-in-a-sentence/solution.cpp
class Solution {
public:
    bool areNumbersAscending(string s) {
        int prev = -1;
        string num = "";
        for (int i = 0; i < s.length(); i++) {
            if (isdigit(s[i])) {
                num += s[i];
            } else if (num != "") {
                int curr = stoi(num);
                if (curr <= prev) {
                    return false;
                }
                prev = curr;
                num = "";
            } else {
                continue;
            }
        }
        if (num != "") {
            int curr = stoi(num);
            if (curr <= prev) {
                return false;
            }
        }
        return true;
    }
};

File: problems/2042-check-if-numbers-are-ascending-in-a-sentence/solution.java
class Solution {
    public boolean areNumbersAscending(String s) {
        int prev = -1;
        String[] tokens = s.split(" ");
        for (String token : tokens) {
            try {
                int num = Integer.parseInt(token);
                if (num <= prev) {
                    return false;
                }
                prev = num;
            } catch (NumberFormatException e) {
                // Not a number, ignore
            }
        }
        return true;
    }
}